Sources/Sentry/SentryHub.m

```diff
@@ -10,65 +10,90 @@
 #import <Sentry/SentryHub.h>
 #import <Sentry/SentryClient.h>
 #import <Sentry/SentryBreadcrumbStore.h>
+#import <Sentry/SentryStackLayer.h>
+#import <Sentry/SentryBreadcrumbTracker.h>
 #else
 #import "SentryHub.h"
 #import "SentryClient.h"
 #import "SentryBreadcrumbStore.h"
+#import "SentrySentryStackLayer.h"
+#import "SentryBreadcrumbTracker.h"
 #endif
 
 @interface SentryHub()
 
-@property (nonatomic, strong) SentryClient *client;
+@property (nonatomic, strong) NSMutableArray<SentryStackLayer *> *stack;
 
 @end
 
 @implementation SentryHub
 
-+ (SentryHub *)defaultHub {
-    static SentryHub *_sharedInstance = nil;
-    static dispatch_once_t oncePredicate;
-    dispatch_once(&oncePredicate, ^{
-        _sharedInstance = [[self alloc] init];
-    });
-    return _sharedInstance;
-}
-
-- (instancetype)initWithClient:(SentryClient *)aClient {
-    self = [super init];
-    if (self) {
-        [self bindClient:aClient];
+- (instancetype)init {
+    if (self = [super init]) {
+        SentryScope *scope = [[SentryScope alloc] init];
+        SentryStackLayer *layer = [[SentryStackLayer alloc] init];
+        layer.scope = scope;
+        [self setStack:[@[layer] mutableCopy]];
     }
     return self;
 }
+/**
+ */
+- (void)setupWithClient:(SentryClient * _Nullable)client {
+    SentryStackLayer *stackLayer = [SentryStackLayer new];
+    stackLayer.scope = [SentryScope new];
+    [self setStack:[@[stackLayer] mutableCopy]];
+}
 
 - (void)captureEvent:(SentryEvent *)event {
-    [self.client sendEvent:event withCompletionHandler:nil];
+    [[self getClient] sendEvent:event scope:[self getStackTop].scope withCompletionHandler:nil];
 }
 
 - (void)addBreadcrumb:(SentryBreadcrumb *)crumb {
-    [self.client.breadcrumbs addBreadcrumb:crumb];
+    [[self getStackTop].scope.breadcrumbs addBreadcrumb:crumb];
```

Comment:
We could also add a helper function `[self getScope]` which does `[[self getStackTop].scope`

Reply:
done